File: savefunds-frontend/src/services/auth.service.ts
import api from './api'

export const authService = {
  async login(email: string, password: string) {
    const response = await api.post('/api/auth/login', { email, password })
    return response.data
  },

  async register(email: string, password: string, nom: string, prenom: string) {
    const response = await api.post('/api/auth/register', { email, password, nom, prenom })
    return response.data
  }
}

File: savefunds-backend/src/main/java/be/profacile/savefunds/domain/service/AuthService.java
package be.profacile.savefunds.domain.service;

import be.profacile.savefunds.api.dto.request.LoginRequest;
import be.profacile.savefunds.api.dto.request.RegisterRequest;
import be.profacile.savefunds.api.dto.response.AuthResponse;
import be.profacile.savefunds.api.dto.response.MessageResponse;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    MessageResponse requestPasswordReset(String email);
}
